crates/rattler_solve/src/resolvo/mod.rs

```diff
@@ -579,10 +579,18 @@ impl super::SolverImpl for Solver {
             .specs
             .iter()
             .map(|spec| {
-                let (name, spec) = spec.clone().into_nameless();
-                let name = name.expect("cannot use matchspec without a name");
-                let name_id = provider.pool.intern_package_name(name.as_normalized());
-                provider.pool.intern_version_set(name_id, spec.into())
+                let (name, nameless_spec) = spec.clone().into_nameless();
+                let name = if let Some(name) = name {
+                    name.as_normalized().to_string()
+                } else if let Some(url) = &nameless_spec.url {
+                    url.to_string()
+                } else {
```

Comment:
I think we should not allow this, because the name of the package and the uri are different we might get into a situation where we have two packages for the same name.